samcli/commands/_utils/options.py

```diff
@@ -156,6 +156,21 @@ def remote_invoke_boto_parameter_callback(ctx, param, provided_value):
     return boto_api_parameters
 
 
+def local_add_host_callback(ctx, param, provided_value):
```

Comment:
I thought custom type should handle this as well, do you know why we needed a callback function here?

Reply:
I have copy/pasted the functionality from the reference `def remote_invoke_parameter_click_option():` and there I saw `RemoteInvokeBotoApiParameterType` is used to parse single argument and `remote_invoke_boto_parameter_callback` callback is utilized to combine them in a dictionary.

If there is a better approach, I can try to refactor it with your guidance.